File: docker/run_docker.py
import os

image_tag = "jdposa/legend_t2dm:0.3"
local_folder = "/home/jposada/Documents/github_repos/LegendT2dm/docker"
gcloud_local_folder = "/home/jposada/.config/gcloud"
user = "ohdsi"
password = "ohdsi"
forwarded_port = 9000



bash_string = f"""
sudo docker run -d \
-v {local_folder}:/workdir/workdir \
-v {gcloud_local_folder}:/workdir/gcloud \
--name=rstudio-ohdsi \
-p {forwarded_port}:8787 \
-e ROOT=TRUE \
-e USER={user} \
-e PASSWORD={password} \
{image_tag}
"""

os.system(bash_string)

File: docker/run_docker.R
image_tag <- "jdposa/legend_t2dm:0.4"
local_folder <-  <YOUR LOCAL FOLDER HERE>
gcloud_local_folder <- <YOUR LOCAL GCLOUD FOLDER HERE>
user <- "ohdsi"
password <- "ohdsi"
forwarded_port <- 9001

bash_string <- sprintf("
sudo docker run -d -v %s:/workdir/workdir -v %s:/workdir/gcloud --name=rstudio-ohdsi -p %s:8787 -e ROOT=TRUE -e USER=%s -e PASSWORD=%s %s",
local_folder,
gcloud_local_folder,
forwarded_port,
user,
password,
image_tag)

print(bash_string)
system(bash_string)
